repaygithub/cactus
# CACTUS-132 :: DataGrid (Not a Draft Anymore)

Ticket: https://repayonline.atlassian.net/browse/CACTUS-132

### Some Notes

- I used the React Context API to help achieve the API structure we defined. Since each column is a child of `DataGrid`, I passed functions in the context down to these children, which are used to push that data back up to the top level `DataGrid` component.
- Developers will be expected to update `paginationOptions` and `sortOptions` and pass them back in through the props; the `DataGrid` does not manage those values internally aside from updating one or two fields at a time and giving the data back to the `onPageChange` or `onSort` handlers respectively.
- I just went ahead and removed `DataGrid` from the exports since this won't be ready to go out until we implement the card view sorting.

### Testing

- Check out this branch & run `yarn web start`
- Make your way to the `DataGrid` storybook and test that the following features have been implemented correctly
    - Sorting based on a column. The sortable columns in the storybook are "Created" and "Active"
    - Pagination. You can toggle the "Provide Page Count" knob to switch between the `Pagination` and `PrevNext` components.
    - Toggle the "Provide Page Size Options" and "Show Results Count" knobs to see what the component looks like with/without those features
    - Change your viewport to test the card view
    - Make sure to test in other browsers such as IE as well!
